replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/StatusCondenser.java

```diff
@@ -0,0 +1,183 @@
+      ReplicatorConfigLoader replicatorConfigLoader,
+      ScheduledExecutorService scheduledExecutor) {
+    this.replicatorHistory = notNull(replicatorHistory);
+    this.replicatorConfigLoader = notNull(replicatorConfigLoader);
+    this.scheduledExecutor = notNull(scheduledExecutor);
+    this.security = Security.getInstance();
+  }
+
+  public StatusCondenser(
+      ReplicatorHistory replicatorHistory,
+      ReplicatorConfigLoader replicatorConfigLoader,
+      ScheduledExecutorService scheduledExecutor,
+      Security security) {
+    this.replicatorHistory = notNull(replicatorHistory);
+    this.replicatorConfigLoader = notNull(replicatorConfigLoader);
+    this.scheduledExecutor = notNull(scheduledExecutor);
+    this.security = security;
+  }
+
+  public void init() {
+    scheduledExecutor.scheduleAtFixedRate(
+        this::condenseResultsAsSystemUser, 1, 60, TimeUnit.MINUTES);
+  }
+
+  public void destroy() {
+    scheduledExecutor.shutdownNow();
+  }
+
+  void condenseResultsAsSystemUser() {
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  replicatorConfigLoader.getAllConfigs().stream().forEach(this::condenseResults);
+                  return null;
+                });
+          } catch (SecurityServiceException | InvocationTargetException e) {
+            LOGGER.error("Error condensing replication history.", e);
+          }
+          return null;
+        });
+  }
+
+  void condenseResults(ReplicatorConfig config) {
+    List<ReplicationStatus> events =
+        Lists.reverse(
+            replicatorHistory
+                .getReplicationEvents(config.getName())
+                .stream()
+                .filter(s -> s.getPeriod().equals(StatusPeriod.INSTANCE))
+                .collect(Collectors.toList()));
+    condenseResults(events, StatusPeriod.DAY, 1);
+    events =
+        Lists.reverse(
+            replicatorHistory
+                .getReplicationEvents(config.getName())
+                .stream()
+                .filter(s -> s.getPeriod().equals(StatusPeriod.DAY))
+                .collect(Collectors.toList()));
+    condenseResults(events, StatusPeriod.WEEK, 7);
+    events =
+        Lists.reverse(
+            replicatorHistory
+                .getReplicationEvents(config.getName())
+                .stream()
+                .filter(s -> s.getPeriod().equals(StatusPeriod.WEEK))
+                .collect(Collectors.toList()));
+    condenseResults(events, StatusPeriod.YEAR, 365);
+  }
+
+  /**
+   * Condenses events into new events of the given type based on the given window
+   *
+   * @param events events to condense. Events must be ordered oldest to newest
+   * @param outType the output event type
+   * @param daysInWindow number of days in the outputType
+   */
+  void condenseResults(List<ReplicationStatus> events, StatusPeriod outType, int daysInWindow) {
+    ReplicationStatus oldest = null;
```

Comment:
✏️ Since this is going to be continuously modified, should we give it another name to better explain what it will hold later? Maybe call it base, condensed, or result.